<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
    
    <head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
        <!-- AppResources meta begin -->
        <script type="text/javascript">var ncbi_startTime = new Date();</script>
        <!-- AppResources meta end -->
        
        <!-- TemplateResources meta begin -->
        <meta name="paf_template" content="" />

        <!-- TemplateResources meta end -->
        
        <!-- Logger begin -->
        <meta name="ncbi_db" content="pmc" /><meta name="ncbi_pdid" content="article" /><meta name="ncbi_acc" content="" /><meta name="ncbi_domain" content="nihpa" /><meta name="ncbi_report" content="record" /><meta name="ncbi_type" content="fulltext" /><meta name="ncbi_objectid" content="" /><meta name="ncbi_pcid" content="/articles/PMC3137130/" /><meta name="ncbi_app" content="pmc" />
        <!-- Logger end -->
        
        <title>An introduction to information retrieval: applications in genomics</title>
        
        <!-- AppResources external_resources begin -->
        <link rel="stylesheet" href="/core/jig/1.14.8/css/jig.min.css" /><script type="text/javascript" src="/core/jig/1.14.8/js/jig.min.js"></script>

        <!-- AppResources external_resources end -->
        
        <!-- Page meta begin -->
        <meta name="robots" content="INDEX,NOFOLLOW,NOARCHIVE" /><link rel="canonical" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137130/" /><link rel="schema.DC" href="http://purl.org/DC/elements/1.0/" /><meta name="citation_journal_title" content="The pharmacogenomics journal" /><meta name="citation_title" content="An introduction to information retrieval: applications in genomics" /><meta name="citation_authors" content="P M Nadkarni" /><meta name="citation_date" content="2002" /><meta name="citation_issue" content="2" /><meta name="citation_volume" content="2" /><meta name="citation_firstpage" content="96" /><meta name="citation_abstract_html_url" content="/pmc/articles/PMC3137130/?report=abstract" /><meta name="citation_pmid" content="12049181" /><meta name="DC.Title" content="An introduction to information retrieval: applications in genomics" /><meta name="DC.Type" content="Text" /><meta name="DC.Publisher" content="NIH Public Access" /><meta name="DC.Contributor" content="P M Nadkarni" /><meta name="DC.Date" content="2002" /><meta name="DC.Language" content="en" /><meta property="og:title" content="An introduction to information retrieval: applications in genomics" /><meta property="og:type" content="article" /><meta property="og:description" content="Information retrieval (IR) is the field of computer science that deals with the processing of documents containing free text, so that they can be rapidly retrieved based on keywords specified in a user’s query. IR technology is the basis of Web-based ..." /><meta property="og:url" content="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137130/" /><meta property="og:site_name" content="PubMed Central (PMC)" /><meta property="og:image" content="https://www.ncbi.nlm.nih.gov/corehtml/pmc/pmcgifs/pmc-logo-share.png" /><meta name="twitter:card" content="summary" /><meta name="twitter:site" content="@ncbi" /><link rel="stylesheet" href="/corehtml/pmc/css/3.14.1/pmc.min.css" type="text/css" /><link rel="stylesheet" href="/corehtml/pmc/css/3.14.1/pmc_extras_prnt.min.css" type="text/css" media="print" /><script type="text/javascript" src="/corehtml/pmc/js/common.min.js">//</script><script type="text/javascript" src="/corehtml/pmc/js/NcbiTagServer.min.js">//</script><meta name="citationexporter" content="backend:'https://api.ncbi.nlm.nih.gov/lit/ctxp/v1/pmc/'" /><script type="text/javascript" src="https://www.ncbi.nlm.nih.gov/corehtml/pmc/ctxp/jquery.citationexporter.min.js">//</script><link rel="stylesheet" href="https://www.ncbi.nlm.nih.gov/corehtml/pmc/ctxp/citationexporter.css" type="text/css" /><script type="text/javascript" src="/core/mathjax/2.6.1/MathJax.js?config=/corehtml/pmc/js/mathjax-config-classic.3.4.js"></script><script type="text/javascript">window.name="mainwindow";</script><style type="text/css">.pmc-wm {background:transparent repeat-y top left;background-image:url(/corehtml/pmc/pmcgifs/wm-hhspa.gif);background-size: auto, contain}</style><style type="text/css">.print-view{display:block}</style><style type="text/css">
        div.pmc_para_cit li.highlight,
        div.pmc_para_cit li.highlight .one_line_source
        { background: #E0E0E0; }
        a.bibr.highlight { background: #E0E0E0; } 
      </style><meta name="cited_in_systematic_reviews" content="" /><link rel="alternate" type="application/epub+zip" href="/pmc/articles/PMC3137130/epub/" /><link rel="alternate" type="application/pdf" href="/pmc/articles/PMC3137130/pdf/nihms303260.pdf" />

        <!-- Page meta end -->
    <link rel="shortcut icon" href="//www.ncbi.nlm.nih.gov/favicon.ico" /><meta name="ncbi_phid" content="8A1A8E52E53566610000000005600560.m_8" />
<meta name='referrer' content='origin-when-cross-origin'/><link type="text/css" rel="stylesheet" href="//static.pubmed.gov/portal/portal3rc.fcgi/4160049/css/3852956/3985586/3808861/4121862/3974050/3917732/251717/4048120/3846471/14534/45193/4113719/3849091/3984811/3751656/4033350/3840896/3577051/3852958/3881636/3579733/4062871/12930/3964959/3855473/4047625/3854974/3854955/4076335/4128070/9685/3549676/3609192/3609193/3609213/3395586/4143404.css" /><link type="text/css" rel="stylesheet" href="//static.pubmed.gov/portal/portal3rc.fcgi/4160049/css/3411343/3882866/4157116.css" media="print" /></head>
    <body class="article">
        <div class="grid">
            <div class="col twelve_col nomargin shadow">
                <!-- System messages like service outage or JS required; this is handled by the TemplateResources portlet -->
                <div class="sysmessages">
                    <noscript>
	<p class="nojs">
	<strong>Warning:</strong>
	The NCBI web site requires JavaScript to function. 
	<a href="/guide/browsers/#enablejs" title="Learn how to enable JavaScript" target="_blank">more...</a>
	</p>
	</noscript>
                </div>
                <!--/.sysmessage-->
                <div class="wrap">
                    <div class="page">
                        <div class="top">
                            <div class="universal_header" id="universal_header"><ul class="inline_list jig-ncbimenu ui-ncbimenu resources_list" id="navcontent"><li class="ui-ncbimenu-item-leaf ui-ncbimenu-item-first ui-helper-reset ui-ncbimenu-item-no-hlt"><a class="ui-ncbimenu-link-first" href="/" role="banner" title="NCBI Home" id="ncbihome" accesskey="1"><span class="offscreen_noflow">NCBI</span><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/28977" class="ncbi_logo" title="NCBI" alt="NCBI Logo" /></a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="#maincontent" title="Skip to the content" tabindex="0" accesskey="3">Skip to main
                        content</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="#navcontent" title="Skip to the navigation" tabindex="0" accesskey="4">Skip to
                        navigation</a></li><li id="resource-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first ui-helper-reset"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="/static/header_footer_ajax/submenu/#resources">Resources</a></li><li id="all-howtos-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="/static/header_footer_ajax/submenu/#howto">How To</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="/guide/browsers/#accesskeys" title="About My NCBI Accesskeys" tabindex="0" accesskey="0">About NCBI Accesskeys</a></li></ul><div class="myncbi"><span id="myncbiusername" style="display:none"><a href="/account/settings/" id="mnu" title="Edit account settings"></a></span><a accesskey="2" href="/myncbi/" id="myncbi" style="display:none">My NCBI</a><a href="/account/" id="sign_in">Sign in to NCBI</a><a href="/account/signout/" id="sign_out" style="display:none">Sign Out</a></div></div>
                            <div class="header">
    <div class="res_logo">
  <h1 class="img_logo"><a href="/pmc/" class="pmc_logo offscreen">PMC</a></h1>
  <div class="NLMLogo">
    <a href="https://www.nlm.nih.gov/" title="US National Library of Medicine">US National Library of Medicine</a>
    <br />
    <a href="https://www.nih.gov/" title="National Institutes of Health">National Institutes of Health</a>
  </div>
</div>
    <div class="search"><form method="get" action="/pmc/"><div class="search_form"><label for="database" class="offscreen_noflow">Search database</label><select id="database"><optgroup label="Recent"><option value="pmc" selected="selected" class="last" data-ac_dict="pmc-search-autocomplete">PMC</option></optgroup><optgroup label="All"><option value="gquery">All Databases</option><option value="assembly">Assembly</option><option value="biocollections">Biocollections</option><option value="bioproject">BioProject</option><option value="biosample">BioSample</option><option value="biosystems">BioSystems</option><option value="books">Books</option><option value="clinvar">ClinVar</option><option value="cdd">Conserved Domains</option><option value="gap">dbGaP</option><option value="dbvar">dbVar</option><option value="gene">Gene</option><option value="genome">Genome</option><option value="gds">GEO DataSets</option><option value="geoprofiles">GEO Profiles</option><option value="gtr">GTR</option><option value="homologene">HomoloGene</option><option value="ipg">Identical Protein Groups</option><option value="medgen">MedGen</option><option value="mesh">MeSH</option><option value="ncbisearch">NCBI Web Site</option><option value="nlmcatalog">NLM Catalog</option><option value="nuccore">Nucleotide</option><option value="omim">OMIM</option><option value="pmc" data-ac_dict="pmc-search-autocomplete">PMC</option><option value="popset">PopSet</option><option value="probe">Probe</option><option value="protein">Protein</option><option value="proteinclusters">Protein Clusters</option><option value="pcassay">PubChem BioAssay</option><option value="pccompound">PubChem Compound</option><option value="pcsubstance">PubChem Substance</option><option value="pubmed">PubMed</option><option value="snp">SNP</option><option value="sparcle">Sparcle</option><option value="sra">SRA</option><option value="structure">Structure</option><option value="taxonomy">Taxonomy</option><option value="toolkit">ToolKit</option><option value="toolkitall">ToolKitAll</option><option value="toolkitbookgh">ToolKitBookgh</option></optgroup></select><div class="nowrap"><label for="term" class="offscreen_noflow" accesskey="/">Search term</label><div class="nowrap"><input type="text" name="term" id="term" title="Search PMC. Use up and down arrows to choose an item from the autocomplete." value="" class="jig-ncbiclearbutton jig-ncbiautocomplete" data-jigconfig="dictionary:'pmc-search-autocomplete',disableUrl:'NcbiSearchBarAutoComplCtrl'" autocomplete="off" data-sbconfig="ds:'no',pjs:'no',afs:'yes'" /></div><button id="search" type="submit" class="button_search nowrap" cmd="go">Search</button></div></div></form><ul class="searchlinks inline_list"><li>
                        <a href="/pmc/advanced/">Advanced</a>
                    </li><li>
                        <a href="/pmc/journals/">Journal list</a>
                    </li><li class="help">
                        <a target="_blank" href="/books/NBK3825/">Help</a>
                    </li></ul></div>
</div>

                            
                            
                        <!--<component id="Page" label="headcontent"/>-->
                            
                        </div>
                        <div class="content">
                            <!-- site messages -->
                            <div class="container">
    <div id="maincontent" class="content eight_col col">
        <div class="navlink-box">
            <ul class="page-breadcrumbs inline_list small"><li class="journal-list"><a href="/pmc/journals/" class="navlink">Journal List</a></li><li class="article-entrez-filter"><a href="/pmc/?term=hhs%20author%20manuscript[filter]" class="navlink">HHS Author Manuscripts</a></li><li class="accid">PMC3137130</li></ul>
        </div>

        <!-- Journal banner -->
        <div class="pmc-page-banner whole_rhythm"><div><img src="/corehtml/pmc/pmcgifs/logo-hhspa.png" alt="Logo of nihpa" usemap="#logo-imagemap" /><map id="logo-imagemap" name="logo-imagemap"><area shape="rect" coords="0,57,255,75" alt="About Author manuscripts" title="About Author manuscripts" href="https://www.ncbi.nlm.nih.gov/pmc/about/authorms.html" ref="https://www.ncbi.nlm.nih.gov/pmc/about/authorms.html" /><area shape="rect" coords="256,57,500,75" alt="Submit a manuscript" title="Submit a manuscript" href="https://www.nihms.nih.gov/" ref="reftype=publisher&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A" target="pmc_ext" /><area shape="rect" coords="0,0,499,74" alt="HHS Public Access; Author Manuscript; Accepted for publication in peer reviewed journal;" title="HHS Public Access; Author Manuscript; Accepted for publication in peer reviewed journal;" href="https://www.ncbi.nlm.nih.gov/pmc/about/public-access/" ref="reftype=publisher&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A" target="pmc_ext" /></map></div> </div>
        
        <!--component id='MainPortlet' label='search-reference'/-->
        
        <!-- Book content -->
        <div class="">
            
        
            
            <div class="hide-overflow article lit-style content pmc-wm slang-all page-box"><!--main-content--><div class="jig-ncbiinpagenav" data-jigconfig="smoothScroll: false, allHeadingLevels: ['h2'], headingExclude: ':hidden'"><div class="fm-sec half_rhythm no_top_margin"><div class="fm-citation half_rhythm no_top_margin clearfix"><div class="inline_block eight_col va_top"><div><span id="pmcmata">Pharmacogenomics J</span>. Author manuscript; available in PMC 2011 Jul 15.</div><div></div><div>Published in final edited form as:</div><div style="margin-left:1em"><div class="fm-vol-iss-date"><a href="/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;retmode=ref&amp;cmd=prlinks&amp;id=12049181" target="pmc_ext" ref="reftype=publisher&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CArticle%7CFree%20Access"><span class="cit">Pharmacogenomics J. 2002; 2(2): 96–102. </span></a></div> </div></div><div class="inline_block four_col va_top show-overflow align_right"><div class="fm-citation-ids"><div class="fm-citation-pmcid"><span class="fm-citation-ids-label">PMCID: </span><span>PMC3137130</span></div><div class="fm-citation-manuscriptid"><span class="fm-citation-ids-label">NIHMSID: </span><span>NIHMS303260</span></div><div class="fm-citation-pmid">PMID: <a href="/pubmed/12049181">12049181</a></div></div></div></div><h1 class="content-title">An introduction to information retrieval: applications in genomics</h1><div class="half_rhythm"><div class="contrib-group fm-author"><a href="/pubmed/?term=Nadkarni%20PM%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=12049181" class="affpopup" co-rid="_co_idm140313716480560" co-class="co-affbox">P M Nadkarni</a><sup>1</sup></div><div style="display:none" class="contrib-group aff-tip"><div id="_co_idm140313716480560"><h3 class="no_margin">P M Nadkarni</h3><p><sup>1</sup> Center for Medical Informatics, Yale University School of Medicine, New Haven, Connecticut, USA</p><div>Find articles by <a href="/pubmed/?term=Nadkarni%20PM%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=12049181">P M Nadkarni</a></div></div></div></div><div class="fm-panel half_rhythm"><div class="togglers"><a href="#" class="pmctoggle" rid="idm140313716398144_ai">Author information</a> <a href="#" class="pmctoggle" rid="idm140313716398144_cpl">Copyright and License information</a> <a href="/pmc/about/disclaimer/">Disclaimer</a></div><div class="fm-authors-info fm-panel hide half_rhythm" id="idm140313716398144_ai" style="display:none"><div class="fm-affl" lang="en" id="A1"><sup>1</sup> Center for Medical Informatics, Yale University School of Medicine, New Haven, Connecticut, USA</div><div id="FN1">Correspondence: PM Nadkarni, Center for Medical Informatics, Yale University School of Medicine, PO Box 208009, New Haven, CT 06520&#x02013;8009, USA Fax: 203&#x02013;764&#x02013;6717 <a href="mailto:dev@null" data-email="ude.elaY@inrakdaN.hsakarP" class="oemail">ude.elaY@inrakdaN.hsakarP</a></div></div><div class="fm-article-notes fm-panel half_rhythm"></div><div class="permissions fm-panel half_rhythm hide" id="idm140313716398144_cpl" style="display:none"><div class="fm-copyright half_rhythm"><a href="/pmc/about/copyright/">Copyright notice</a> </div></div></div><div id="pmclinksbox" class="links-box whole_rhythm"><div class="fm-panel"><div>The publisher's final edited version of this article is available free at <a href="/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;retmode=ref&amp;cmd=prlinks&amp;id=12049181" target="pmc_ext" ref="reftype=publisher&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CArticle%7CFree%20Access">Pharmacogenomics J</a></div></div></div></div><div class="sec"></div><div id="idm140313717521776" lang="en" class="tsec sec"><h2 class="head no_bottom_margin" id="idm140313717521776title">Abstract</h2><!--article-meta--><div><p id="P1" class="p p-first-last">Information retrieval (IR) is the field of computer science that deals with the processing of documents containing free text, so that they can be rapidly retrieved based on keywords specified in a user&#x02019;s query. IR technology is the basis of Web-based search engines, and plays a vital role in biomedical research, because it is the foundation of software that supports literature search. Documents can be indexed by both the words they contain, as well as the concepts that can be matched to domain-specific thesauri; concept matching, however, poses several practical difficulties that make it unsuitable for use by itself. This article provides an introduction to IR and summarizes various applications of IR and related technologies to genomics.</p></div><div class="sec"><strong class="kwd-title">Keywords: </strong><span class="kwd-text">information retrieval, full-text indexing, text processing, genomics</span></div></div><div id="S1" class="tsec sec"><h2 class="head no_bottom_margin" id="S1title">INTRODUCTION AND HISTORY</h2><p id="P2" class="p p-first">Information Retrieval (IR), (more precisely, <em>text</em> information retrieval) is a branch of computer science that deals with the processing of collections of documents containing &#x02018;free text&#x02019;, such as scientific papers, or even the contents of electronic textbooks. The objective of such processing is to facilitate rapid and accurate search of the text based on keywords of interest. The word &#x02018;free&#x02019; (as opposed to &#x02018;structured&#x02019;) implies that in contrast to, say, an electronic spreadsheet, such documents have little structure that could serve as a guide locating specific content. The term IR originally referred to &#x02018;information&#x02019; in a very broad sense (and dealt with problems such as lossless data transmission and data compression), but the focus on textual information can be traced to several researchers, most notably Salton, van Rijsbergen and Sparck-Jones. The first two have written historically important IR texts;<sup><a href="#R1" rid="R1" class=" bibr popnode">1</a>,<a href="#R2" rid="R2" class=" bibr popnode">2</a></sup> modern IR texts<sup><a href="#R3" rid="R3" class=" bibr popnode">3</a>,<a href="#R4" rid="R4" class=" bibr popnode">4</a></sup> provide an excellent overview. IR overlaps several other fields within computer science, notably database technology and natural language processing (NLP).</p><p id="P3" class="p">For several decades IR was an &#x02018;orphan&#x02019; technology researched by a relative handful of scientists, and commercial offerings were sparse and modestly featured. However, due to the spread of the World Wide Web, IR is now mainstream because most of the information on the Web is textual. Web search engines such as Yahoo, Excite, Alta Vista and AskJeeves are used by millions of users to locate information on Web pages across the world on any topic. The use of search engines has spread to the point where, for people with access to the Internet, the World Wide Web has replaced the library as the reference tool of first choice. Of the individual WWW text-based resources available to the genomics researcher, the single most useful resource is unquestionably PubMed, maintained by the National Center for Biotechnology Information (NCBI), part of the US National Library of Medicine (NLM). PubMed provides access to the contents of MEDLINE, which is curated by the NLM.</p><div id="S2" class="sec"><h3 id="S2title">Document Indexing</h3><p id="P4" class="p p-first-last">There are several ways to pre-process documents electronically so as to speed up their retrieval. All of these fall under the general term &#x02018;<em>indexing</em>&#x02019;: an index is a structure that facilitates rapid location of items of interest, an electronic analog of a book&#x02019;s index. The most widely used technique is <em>word indexing</em>, where the entries (or <em>terms</em>) in the index are individual words in the document (ignoring &#x02018;stop words&#x02019;&#x02014;very common and uninteresting words such as &#x02018;the&#x02019;, &#x02018;an&#x02019;, &#x02018;of&#x02019;, etc). Another technique is <em>concept indexing</em>, where one identifies words or phrases and tries to map them to a thesaurus of synonyms as <em>concepts</em>. Therefore, the terms in the index are concept IDs. Several kinds of indexes are created. The <em>global term-frequency index</em> records how many times each distinct term occurs in the entire document collection. The <em>document term-frequency index</em> records how often a particular term occurs in each document. An optional <em>proximity index</em> records the position of individual terms within the document as word, sentence or paragraph offsets. (A proximity index is voluminous, but can allow searches requiring that two or more terms be within the same sentence, or within so many words of each other.) Some issues in indexing are now discussed.</p></div><div id="S3" class="sec"><h3 id="S3title">Word Indexing</h3><ul class="unordered" style="list-style-type:disc" id="L1"><!--
list-behavior=unordered
prefix-word=
mark-type=disc
max-label-size=0
--><li><p id="P5">In specific contexts, the list of stop words may need to be expanded. Thus &#x02018;drug&#x02019; would be a stop word in a pharmacology-related collection</p></li><li><p id="P6">To reduce the size of the index and to enable broader searches, words may be transformed prior to indexing. One transformation is <em>normalization</em>, which involves lower-case conversion and removal of variations in person or tense. Thus, &#x02018;children&#x02019; is normalized to &#x02018;child&#x02019;, and &#x02018;brought&#x02019; to &#x02018;bring&#x02019;. Another is <em>stemming</em>, which transforms the word by stripping off suffixes or prefixes (using simple pattern-matching rules) to a &#x02018;root&#x02019; form, which is not necessarily an actual word in the language. Stemming can be drastic (and its use therefore controversial). Thus, with the widely used Porter stemming algorithm,<sup><a href="#R5" rid="R5" class=" bibr popnode">5</a></sup> &#x02018;modulation&#x02019; and &#x02018;module&#x02019; yield the same root, &#x02018;modul&#x02019;. The two words mean different things (though they were both originally derived from the Latin term &#x02018;modulus&#x02019;). Conversely, &#x02018;modular&#x02019;, the adjectival form of &#x02018;module&#x02019;, stems to itself, rather than to &#x02018;modul&#x02019;. Several studies of stemming have cast doubt on its utility for English text.<sup><a href="#R6" rid="R6" class=" bibr popnode">6</a>,<a href="#R7" rid="R7" class=" bibr popnode">7</a></sup></p></li></ul></div><div id="S4" class="sec sec-last"><h3 id="S4title">Concept Indexing</h3><p id="P7" class="p p-first">Some practical problems faced in automated concept indexing are discussed.<sup><a href="#R8" rid="R8" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562119">8</a></sup></p><ul class="unordered" style="list-style-type:disc" id="L2"><!--
list-behavior=unordered
prefix-word=
mark-type=disc
max-label-size=0
--><li><p id="P8">The thesaurus may be general-purpose (eg, applying to the English language) or domain-specific. The National Library of Medicine&#x02019;s Unified Medical Language System (UMLS) (which is a <em>meta-thesaurus</em> or conglomeration of the contents of several biomedical thesauri) is the world&#x02019;s largest domain-specific thesaurus. The largest and possibly the best curated of the individual component vocabularies within UMLS is the Medical Subject Headings (MeSH) developed at the NLM for the Index Medicus. The UMLS contains much information that is useful to the software developer. Thus, a concept can belong to one or more <em>semantic categories</em> (eg, pharmacological substance, therapeutic procedure) and every term for a concept is tagged with the <em>ID of the source vocabulary</em> from which it was taken. This information allows researchers to create UMLS data subsets for special purposes. The inter-relationships of concepts are also recorded so that concepts can be arranged in a <em>hierarchy</em> from super-concepts (more general concepts) to more specific/particular concepts. The coverage of the UMLS is not uniform across all fields of biomedicine; coverage depends on the source vocabularies that feed into it. Thus, coverage of clinical concepts is much more extensive than coverage of basic science. The UMLS is therefore a continual work in progress, with major updates being issued annually</p></li><li><p id="P9">Recognition of concepts by scanning of text is quite difficult. Most attempts have focused on identifying individual noun phrases,<sup><a href="#R9" rid="R9" class=" bibr popnode">9</a>&#x02013;<a href="#R12" rid="R12" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_415135619">12</a></sup> though verb phrases such as &#x02018;mutated&#x02019; and &#x02018;catalyzed&#x02019; are also important. The challenge here is a disparity between the granularity of concepts as expressed in phrases in text and as they occur in the thesaurus. Many common concepts in biomedicine are <em>compound</em>, that is, they have been formed by fusion of other concepts, but the extent of fusion is variable. A single phrase may thus correspond to multiple concepts, and vice versa. For example, &#x02018;digitalis-induced atrial fibrillation&#x02019; does not match a single UMLS concept, though the sub-phrases &#x02018;digitalis&#x02019; and &#x02018;atrial fibrillation&#x02019; do. Conversely, the two (underlined) noun phrases in the sentence fragment, &#x02018;<span class="underline">hypertension</span> is secondary to <span class="underline">renal disease</span>&#x02019;, together happen to correspond to a single compound concept, renal hypertension, composed of the two simpler underlined concepts</p></li><li><p id="P10">Matching a phrase to a concept in the thesaurus is also complicated by <em>polysemy</em>: the existence of terms with multiple meanings, or homonyms. Thus, to cite an example from a recent review by Daniel Masys,<sup><a href="#R13" rid="R13" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_325405919">13</a></sup> &#x02018;insulin&#x02019; could refer to the gene, the protein, the hormone, or therapy. Neighboring words may help to disambiguate, eg, the phrase &#x02018;insulin molecule&#x02019; can be uniquely matched. However, if the disambiguating word is further away, eg in the previous or succeeding sentence, the problem is much more difficult</p></li><li><p id="P11">Every field evolves its own set of <em>acronyms</em>, and the same acronym may appear in different fields even though they do not overlap in their subject matter. Thus, it is possible for a bioinformatics article to cite the Microsoft acronym DNA (&#x02018;distributed network architecture&#x02019;, a term coined by Bill Gates), rather than the expected biological equivalent (&#x02018;deoxyribonucleic acid&#x02019;)</p></li><li><p id="P12">The concept&#x02019;s presence in a document does not, <em>per se</em>, make that document relevant for that concept, because the concept may be negated, or ruled out. A paper by Mutalik <em>et al</em><sup><a href="#R14" rid="R14" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562113">14</a></sup> showed that concept negation in <em>dictated</em> medical narrative (such as operative notes or discharge summaries) could mostly be detected reliably by simple computational strategies. This conclusion, however, does not apply generally to biomedical literature. When a trained (and busy) clinician dictates notes, the goal is to convey information directly and succinctly. Spoken negations tend to be simple in structure, with the negating phrase being present mostly within the same sentence as the negated concept, eg, &#x02018;no history of vomiting or diarrhea&#x02019;. In the written word, such directness is uncommon, and the negation often lies in the next sentence or paragraph (as when a hypothesis is described, only to be refuted later). Further, written medical text is full of &#x02018;linguistic hedges&#x02019; that language purists loathe, such as &#x02018;not uncommon&#x02019; (which is intended to convey a different meaning than &#x02018;common&#x02019;)</p></li></ul><p id="P13" class="p p-last">Because of the issues outlined above, concept indexing can, at best, serve only as ancillary to word indexing instead of as a replacement.</p></div></div><div id="S5" class="tsec sec"><h2 class="head no_bottom_margin" id="S5title">QUERYING TEXT DATABASES: RELEVANCE RANKING AND CLUSTERING</h2><p id="P14" class="p p-first">An important feature of search using IR technology is that when the user requests documents containing terms of interest, some non-relevant documents may be returned (false positives) and some relevant documents may be missed (false negatives). Errors are due to artifacts of the indexing/concept recognition process. One can therefore conceive of the sensitivity and specificity of a search: the equivalent terms in the IR literature are <em>recall</em> and <em>precision</em> respectively. Further, since the same term can occur multiple times in the same document, when searching a large document database, a simple Boolean query (&#x02018;return all documents containing terms X, Y and Z&#x02032;) is not very useful if it returns thousands of documents that contain these terms even once. The search must therefore be augmented using <em>relevance ranking</em>, where documents that provide the &#x02018;best match&#x02019; are ranked highest.</p><p id="P15">Relevance ranking is typically done through variants of the <em>document vector approach</em>,<sup><a href="#R15" rid="R15" class=" bibr popnode">15</a>,<a href="#R16" rid="R16" class=" bibr popnode">16</a></sup> which uses the frequency indexes mentioned earlier. Terms in the query that are uncommon in the document collection are given more weight than relatively common terms, and documents containing one or more of the query terms many times (in proportion to the document&#x02019;s length) are weighted more than documents containing the terms infrequently or not at all. All mainstream text-search systems, including PubMed, use the document vector method. Other customized weightings may be used for scientific articles: eg, terms in the title and abstract are weighted more than terms in the body of the text. Note that relevance ranking can return useful results even if no document contains <em>every</em> single term in the user&#x02019;s query. One can use Boolean criteria to constrain the search, specifying, for example that certain terms in the query <em>must</em> (or must not) be present in a document for it to be considered relevant. A useful overview of weighting approaches is given in Sparck Jones <em>et al</em>.<sup><a href="#R17" rid="R17" class=" bibr popnode">17</a>,<a href="#R18" rid="R18" class=" bibr popnode">18</a></sup></p><p id="P16" class="p">As an aside, one must add that today&#x02019;s commercial engines for Web-page indexing use various proprietary refinements in their relevance-ranking algorithms. The earlier approaches, which were based strictly on frequency indexes, could easily be fooled by Web pages that repeated the same word (such as an obscene phrase) hundreds of times within the HTML page as a comment. (HTML comments are invisible to the human viewer of the page, but visible to the computer programs that perform the indexing.) As a result, the relevancy-ranking program would rank such pages highly when a search using one of those words was performed. Modern engines are harder to trick. Google for example, uses a weighting algorithm based on how many other Web pages in the collection refer to a particular page by way of hyperlinks.<sup><a href="#R19" rid="R19" class=" bibr popnode">19</a></sup> In a sense, the other pages &#x02018;vote&#x02019; for some pages as being more useful/relevant than others. Further, links from pages that are themselves important based on the votes they have received are weighted more than links from non-highly-ranked pages.</p><div id="S6" class="sec sec-last"><h3 id="S6title">Other Uses of the Document Vector Approach</h3><p id="P17" class="p p-first">One can use the document vector approach to derive a similarity measure between two documents, based on the terms that they have in common (and their relative frequency of occurrence). This information can be pre-computed when documents are first loaded into the database. This is what NCBI&#x02019;s Entrez system (which accesses several databases, including PubMed) does. Every document is compared with every other previously loaded document in the database, using an algorithm devised by NCBI&#x02019;s John Wilbur, and the highest matches are stored for future retrieval. Instant access to the highest matches is available to PubMed&#x02019;s users with the &#x02018;Find Related Articles&#x02019; hyperlink. The similarity algorithm is quite powerful, and it has been successfully used for a purpose not originally intended by Wilbur. The journal <em>Science</em> reported that a researcher using PubMed identified a serial plagiarist.<sup><a href="#R20" rid="R20" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562117">20</a></sup> The &#x02018;Find Related&#x02019; link, applied to several of the suspect&#x02019;s articles in turn, identified the original article (whose contents were plagiarized) correctly in every case at the top of the &#x02018;highly similar articles&#x02019; list. Even more surprising, the plagiarized articles had been written in a different language, and Wilbur&#x02019;s algorithm was working with English translations of the abstracts.</p><p id="P18" class="p p-last">The document-vector approach can also be used to automatically cluster similar documents in a collection, and the clusters that are identified can then be inspected and interpreted. A new document can then be classified electronically into one of these clusters, based on which documents it most resembles. Automatic document clustering has been used for various forms of &#x02018;literary detective work&#x02019;. Term frequency does not have to be the only criterion for quantifying similarity. Other measures, such as sentence length, the proportion of verbs with a passive voice, the higher-order semantic categories to which individual terms belong, and so on, can also be utilized in the weighting scheme, to create a kind of &#x02018;literary fingerprint&#x02019; for the author of the work. This information can indicate if a well-known author wrote a recently discovered unknown work, and can help identify possible literary forgeries. In both these cases, comparison of various similarity measures is done using a collection of the author&#x02019;s existing works (as well as the works of several other authors, to serve as controls). During the Cold War, CIA &#x02018;Kremlin watchers&#x02019; used document clustering to identify particular communiqu&#x000e9;s published in Pravda as having come from one of several (anonymous) political commentators.</p></div></div><div id="S7" class="tsec sec"><h2 class="head no_bottom_margin" id="S7title">IR AND TRADITIONAL DATABASES</h2><p id="P19" class="p p-first">Today, IR technology is available as part of standard relational database management system (RDBMS) offerings of all major vendors. This enables integrated query of databases that contain a mix of structured and unstructured data, and offers a variety of methods for retrieving information from databases. (It is rather surprising that the integration has taken this long: IR indexes can in fact be implemented with RDBMS technology.) These offerings, however, currently lag somewhat behind those of vendors specializing in IR. For example, proximity indexing and synonym indexing (based on an English-language thesaurus) are currently unavailable with RDBMS-based offerings. Specialized IR systems, such as dtSearch<sup>&#x02122;</sup> will also index text in a larger variety of formats (word-processing, PDF, HTML) than their RDBMS counterparts. (dtSearch (<a href="http://www.dtsearch.com" data-ga-action="click_feat_suppl" ref="reftype=extlink&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CBody&amp;TO=External%7CLink%7CURI" target="_blank">www.dtsearch.com</a>) is the engine used to index the widely distributed Physician&#x02019;s Desk Reference (PDR) pharmacotherapy guide, which is distributed on CD-ROM.) The difference between RDBMS and &#x02018;pure IR&#x02019; vendors is expected, however, to shrink with time, although, because of the pressure on the latter to innovate in order to survive (IR vendors are much smaller than RDBMS vendors in terms of gross revenues), it will probably never shrink to the point of insignificance.</p><p id="P20" class="p p-last">The degree of IR integration, and ease of use, depends on the RDBMS vendor. Microsoft SQL Server 2000 provides an example of high-quality integration. When indexing a particular table containing numerous fields, the database administrator simply picks all text fields that are considered worth indexing through a graphical user interface tool (a &#x02018;wizard&#x02019;), and the engine does the rest. MS SQL Server enables search of the data using proprietary extensions of SQL (Structured Query Language, the lingua franca for RDBMS data manipulation). Four programming functions, whose documentation is about three pages long in total, have been added to Microsoft&#x02019;s version of the language, allowing Boolean-style query as well as relevance ranking All of this can be learned and used by most programmers with an hour or two of effort. When searching for a key phrase, one can simply specify that <em>all</em> indexed fields must be searched, so that Google-like user interfaces can be programmed quite easily.</p></div><div id="S8" class="tsec sec"><h2 class="head no_bottom_margin" id="S8title">APPLICATIONS IN GENOMICS</h2><div id="S9" class="sec sec-first"><h3 id="S9title">Delivery of Databases Dealing with Specialized Content over the Web</h3><p id="P21" class="p p-first">Many databases in genomics contain a large amount of textual content. Some, such as OMIM<sup><a href="#R21" rid="R21" class=" bibr popnode">21</a></sup> (Online Mendelian Inheritance in Man), which is curated by Victor McKusick&#x02019;s group at Johns Hopkins University and published over the Web by NCBI, contain text predominantly. An annual issue of <em>Nucleic Acids Research</em> provides brief papers describing various genomics-related databases, and practically all of these are accessible via the Web. Even when a database might really be non-textual and supported by a relational engine, users are so used to keyword-based search (without needing to specify what field the key phrase lies in), that the use of IR technology becomes almost mandated. Vendors of IR engines, such as Google and AskJeeves, provide a valuable service to the WWW community by providing their indexing facilities free of charge. Ultimately, however, their objective is to sell their engines to customers who need to provide similar facilities for their own Web sites. For those customers with more modest budgets, however, less full-featured indexing engines are also bundled with the Microsoft, Netscape and Apache Web servers.</p><p id="P22" class="p p-last">Many genomics-related databases consist almost entirely of text files and HTML pages with hyperlinks.<sup><a href="#R22" rid="R22" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562112">22</a>,<a href="#R23" rid="R23" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562116">23</a></sup> In such a case, often the only off-the-shelf software technology that is needed is Web-page indexing capability. For example, GeneCards,<sup><a href="#R24" rid="R24" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562118">24</a></sup> built by a group at Israel&#x02019;s Weizmann Institute and mirrored at several sites such as the National Cancer Institute, uses the Excite engine to facilitate full-text search. GeneCards collates information on individual genes from several sources, such as SWISS-PROT, OMIM and GDB and also allows &#x02018;approximate searching&#x02019;<sup><a href="#R25" rid="R25" class=" bibr popnode">25</a></sup> based on the gene symbol. In approximate searching, a phrase of interest can be located in a block of text even if a certain proportion of letters in a search pattern are mistyped or omitted. It is particularly useful for gene symbols because many symbols are highly similar, differing only with respect to numeric suffixes (and because symbols are often revised subtly).</p></div><div id="S10" class="sec"><h3 id="S10title">Data Mining With IR Technologies</h3><p id="P23" class="p p-first">&#x02018;Data mining&#x02019; refers to automated discovery of information by electronic processing of data (typically, large data volumes) that may not have been expressly gathered for that purpose. While some authors use the phrase <em>knowledge discovery</em>, &#x02018;knowledge&#x02019; is a somewhat more pretentious term than &#x02018;information&#x02019;. Data mining programs only detect patterns that might possibly be useful in confirming hypotheses or generating new, possibly interesting hypotheses. This information can only be elevated to the level of &#x02018;knowledge&#x02019; if and when it proves to be useful. Many detected patterns might in fact be self-obvious to someone who is intimately, or even superficially, familiar with the nature of the data that is being mined. An obvious example is mining of medical data, which discovers the pattern, which is 100% specific, that ovarian cancer occurs only in females.</p><p id="P24">Data mining uses both classical and modern statistical algorithms. (One of the major vendors in the multi-million-dollar data mining industry is the SAS Institute, vendor of the SAS statistics package.) It also makes use of more flexible, but far more computationally intensive approaches that make fewer assumptions about the nature of the data distribution for individual parameters&#x02014;such as neural networks. However, for special kinds of data, almost any ad hoc analysis can be performed with specially written programs, as long as the researchers are able to justify their approach.</p><p id="P25" class="p p-last">The phrase &#x02018;text mining&#x02019; originated as a marketing term by IR vendors who saw that, when the data being explored is primarily textual, traditional IR technologies such as automatic clustering could be productively deployed to satisfy information hunger. In many cases, &#x02018;text mining software&#x02019; was little more than repackaging of existing IR software. In the genomics area, however, innovative algorithms that could truly be described as mining of text have produced interesting results. We describe a few of these here: this sample is biased by the reviewer&#x02019;s interest in applications that combine the use of PubMed with components of the UMLS, Genbank and other publicly accessible databases.</p></div><div id="S11" class="sec"><h3 id="S11title">Using Concept Hierarchies with PubMed to Assist Interpretation of Microarray-based Expression Patterns</h3><p id="P26" class="p p-first">In microarray-based experiments, a particular experimental intervention (eg, the administration of a chemical) results in a decreased or increased expression of several genes. After analysis of the expression pattern using several standard statistical approaches, these genes can be grouped into clusters. However, trying to find a biological basis for the existence of a particular observed cluster is a more complicated task. While the standard textual descriptions associated with gene names include keywords such as &#x02018;respiration&#x02019;, &#x02018;metabolism&#x02019; and so on, these are typically so broad as to be only minimally informative in interpretation: the best guide is still the current scientific literature.</p><p id="P27">Simply returning the hundreds of citations that would apply to the collective set of genes in a cluster is not enough, however. One must attempt to inter-relate and organize these in some way. The program devised by Daniel Masys and his colleagues at the University of California, San Diego<sup><a href="#R26" rid="R26" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562115">26</a></sup> accepts a list of up to 750 gene identifiers (which form the individual components of a cluster). For each gene, the top 20 PubMed citations for each identifier are retrieved (if available: Masys&#x02019; paper states that currently, less than half the genes on commercial arrays have associated publications, because many of these are anonymous ESTs). MeSH terms and chemical registry number (RN) keywords are extracted from the citations and matched to UMLS concept IDs. Where the concepts refer to enzymes, EC numbers are also retrieved. The UMLS concept hierarchy is then used to retrieve all the ancestral (ie, more general) concepts corresponding to the retrieved concepts.</p><p id="P28">A set of &#x02018;concept hierarchy trees&#x02019; is now generated as an HTML document. Each tree corresponds to a broad semantic category as defined in the UMLS (eg, enzymes, diseases), with more general concepts higher up in the tree. Each item on a tree is a key phrase (eg, &#x02018;Cathepsin D&#x02019;, &#x02018;myeloid leukemia&#x02019;, &#x02018;carboxypeptidases&#x02019;) that is labeled with a hyperlink that shows the number of hits. Navigating to the hyperlink reveals the individual genes that matched this particular key phrase, and the associated PubMed citations. Each item is also labeled with a <em>P</em> value that indicates the probability that this keyword would appear purely by chance. (These are computed on the basis of a prior simulation that uses numerous randomly generated gene clusters, and calculates the expected frequencies of individual keywords when generated by chance). The most significant <em>P</em> values, understandably, are those associated with more specific concepts at lower levels of each tree.</p><p id="P29" class="p p-last">The High-Density Array Pattern Interpreter (HAPI), as the program suite described above is called, is publicly accessible via <a href="http://http:www.array.ucsd.edu/hapi/" data-ga-action="click_feat_suppl" ref="reftype=extlink&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CBody&amp;TO=External%7CLink%7CURI" target="_blank">http:www.array.ucsd.edu/hapi/</a>.</p></div><div id="S12" class="sec"><h3 id="S12title">Using Natural Language Processing (NLP) with PubMed to Bootstrap the Contents of Databases or Enhance the Specificity of PubMed Searches</h3><p id="P30" class="p p-first">PubMed does not maintain proximity indexes. As stated in the introduction to this review, the disk space requirements for proximity indexes are formidable: these take up much more space than the documents themselves. The most common form of &#x02018;proximity&#x02019; search is when users wish to search for a phrase exactly as typed. PubMed will perform automatic term mapping by consulting, in succession, the Medical Subject Headings (MeSH) Translation Table, a Journals Translation Table, a Phrase List, and an Author Index.<sup><a href="#R27" rid="R27" class=" bibr popnode">27</a></sup> While this approach maximizes the utility of PubMed for the end-user, the lack of proximity indexing limits its direct use for special-purpose searches. However, there is nothing to prevent one from processing PubMed output after identifying likely candidates using a non-proximity search. (For that matter, since the contents of PubMed are provided at no cost to academic institutions after signing a license agreement, one can, if one dares, proximity-index PubMed using a commercial program if disk space is not a constraint.)</p><p id="P31">Researchers at the National Institutes of Health have developed interesting applications that apply NLP to the contents of PubMed searches. The process is augmented by the use of controlled vocabularies (such as the UMLS, which also contain other useful information such as lists of drugs) or special-purpose databases, such as those containing data on known genes. We briefly describe a few of these.</p><p id="P32"><span class="underline">MedMiner</span><sup><a href="#R28" rid="R28" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562121">28</a></sup> assists interpretation of microarray-based gene expression data by seeking out the appropriate literature. Given a list of genes of interest (or phrases indicative of function, such as &#x02018;apoptosis&#x02019;), MedMiner first queries GeneCards to get a list of all related genes, including those not on the chip. It then retrieves, from PubMed, citations related to those genes. Now, it filters the citations by first parsing the text (title + abstract) of each citation into its individual sentences. It then look for sentences containing at least one gene name/symbol and one or more items from a list of 51 &#x02018;keywords&#x02019; (such as &#x02018;inhibits&#x02019;, &#x02018;stimulates&#x02019;, &#x02018;resistant&#x02019;, and so on, with alternative word forms such as &#x02018;inhibited&#x02019; being handled by stemming). All citations where no such sentences are present are eliminated, and the remaining citations are relevance-ranked by the number of such sentences, and presented to the user.</p><p id="P33">ARBITER<sup><a href="#R29" rid="R29" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_415135620">29</a></sup> extracts information on macromolecular binding relationships from PubMed citations, matching noun phrases to gene and molecule names from Genbank and UMLS, and looking for variants of the verb &#x02018;bound&#x02019;/&#x02019;bind&#x02019; that apply to those phrases as subject and object (eg, X binds Y, X is bound by Y), or noun variants of &#x02018;bind&#x02019;, such as &#x02018;binding of&#x02019;. It also uses a list of phrases from the domain of molecular biology that indicate entities participating in binding (eg, box, sequence, strand, motif, receptor, domain, target). A similar and more recent program, EDGAR (Extraction of Drugs, Genes and Relations)<sup><a href="#R30" rid="R30" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_415135618">30</a></sup> extracts information about drugs and genes applicable to cancer.</p><p id="P34" class="p p-last">Both the ARBITER and EDGAR approaches could be used to automatically populate the contents of special-purpose databases from the biomedical literature. The reliability of automatic database population depends greatly on the precision and recall of the technique used to mine the literature. (The early evaluation of ARBITER describes a precision and recall of 79 and 72 per cent respectively.) If such approaches were sufficiently refined, however, it would be quicker and simpler for human experts to manually curate the contents of an already-populated database (by removing errors due to false positive matches), than having to make entries into a database that was initially empty.</p></div><div id="S13" class="sec sec-last"><h3 id="S13title">Mining PubMed for Suggestion of Research Hypotheses</h3><p id="P35" class="p p-first">Arrowsmith<sup><a href="#R31" rid="R31" class=" bibr popnode">31</a>,<a href="#R32" rid="R32" class=" bibr popnode">32</a></sup> created by Don Swanson and refined with the help of his collaborator Neil Smallheiser, performs hypothesis discovery by an algorithm that is astonishing in its simplicity, which is best described by a scenario. Suppose a patient who is receiving a recently introduced drug is found to have a rare adverse event. The question is whether the agent, rather than the underlying disease, could have been responsible. A PubMed search specifying both the drug and the adverse event returns no references. Suppose we then do a search specifying only the drug, and store all the titles of the returned citations. We do the same with the adverse event. We then eliminate stop-words from each set of titles, put all the remaining words into two sorted lists. We then pick the words/phrases common to both lists. These phrases act as conceptual &#x02018;bridges&#x02019; between the drug and the adverse event. While most bridging phrases may be obvious to the researcher in the context of the domain being investigated, sometimes one may discover unexpected associations, some of which may suggest a cause-and-effect relationship when one goes back to citations containing both the term of interest and the bridging phrase. (Obviously, such relationships must be confirmed by experimental research.)</p><p id="P36">In 1990, Swanson used this approach to discover an unexpected link between dietary magnesium deficiency and some types of migraine.<sup><a href="#R33" rid="R33" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_227562120">33</a></sup> The bridging phrase was &#x02018;cortical depression.&#x02019; On inspecting the citations with &#x02018;magnesium&#x02019; and &#x02018;cortical depression&#x02019;, he discovered that cortical spreading depression was inhibited by magnesium. On looking up &#x02018;migraine&#x02019; and &#x02018;cortical depression&#x02019;, he found that cortical spreading depression was one of the mechanisms involved in some forms of migraine (as well as epilepsy). This suggested that insufficient magnesium predisposed to migraine through the intermediary of cortical spreading depression, a finding that has since been abundantly confirmed.</p><p id="P37" class="p p-last">Arrowsmith, which is publicly accessible via kiwi.uchicago.edu, takes two text files containing the results of PubMed title searches. After a short time, it then returns to the user a list of words and phrases that are common to the two sets of titles. There are various possible refinements to the basic Arrowsmith algorithm. For example, one could increase recall by using the abstracts, or even full text, rather than titles alone; this would also have the unfortunate side effect of increasing the amount of material to be sifted through by the human reviewing the results. One could also make the output more succinct by matching with concepts rather than words to address the synonym problem. Arrowsmith has recently received NIH funding as part of the Human Brain Project, which focuses on Neuro-informatics. We have mentioned Arrowsmith here because it is very likely to be of use in the next phase of the human genome project&#x02014;the task of assigning biological function to all those millions of sequences.</p></div></div><div id="S14" class="tsec sec"><h2 class="head no_bottom_margin" id="S14title">CONCLUSIONS</h2><p id="P38" class="p p-first-last">The most widely available &#x02018;information&#x02019; in today&#x02019;s world is still textual, and IR techniques are broadly applicable in facilitating its productive use. The general availability of large databases such as PubMed, as well as large vocabularies such as the UMLS has greatly facilitated the work of IR researchers in devising and testing scalable approaches that can positively impact the work of the laboratory researcher.</p></div><div id="S15" class="tsec sec"><h2 class="head no_bottom_margin" id="S15title">Acknowledgments</h2><div class="sec"><p id="__p1">The author thanks Cynthia Brandt, MD, and John Fisk, MD, of the Yale Center for Medical Informatics, and the anonymous reviewers for feedback on the article. The author is supported by grants U01 ES10867&#x02013;02 from the National Institute of Environmental Health Sciences, R01 LM06843&#x02013;02 from the National Library of Medicine and U01 CA78266&#x02013;04 from the National Cancer Institute.</p></div></div><div id="idm140313711629856" class="tsec sec"><h2 class="head no_bottom_margin" id="idm140313711629856title">Footnotes</h2><!--back/fn-group--><div class="fm-sec half_rhythm small"><p class="fn sec" id="FN2"> <p id="__p2" class="p p-first"><strong>DUALITY OF INTEREST</strong></p><p id="__p3" class="p p-last">None declared.</p></p></div></div><div id="idm140313717280128" class="tsec sec"><h2 class="head no_bottom_margin" id="idm140313717280128title">References</h2><div class="ref-list-sec sec" id="reference-list"><div class="ref-cit-blk half_rhythm" id="R1">1. <span class="element-citation">Salton G.  <span class="ref-journal">Automatic Text Processing: the transformation, analysis, and retrieval of information by computer.</span> Addison-Wesley; Reading, MA: 1989.  <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?title=Automatic+Text+Processing:+the+transformation,+analysis,+and+retrieval+of+information+by+computer&amp;author=G+Salton&amp;publication_year=1989&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R2">2. <span class="element-citation">Van Rijsbergen CJ.  <span class="ref-journal">Information Retrieval.</span> Butterworths; London, UK: 1979.  <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?title=Information+Retrieval&amp;author=CJ+Van+Rijsbergen&amp;publication_year=1979&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R3">3. <span class="element-citation">Baeza-Yates R, Ribeiro-Neto B.  <span class="ref-journal">Modern Information Retrieval.</span> Addison-Wesley Longman; Harlow, UK: 1999.  <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?title=Modern+Information+Retrieval&amp;author=R+Baeza-Yates&amp;author=B+Ribeiro-Neto&amp;publication_year=1999&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R4">4. <span class="element-citation">Witten IH, Moffat A, Bell TC.  <span class="ref-journal">Managing Gigabytes.</span> Morgan Kaufman; San Francisco, CA: 1999.  <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?title=Managing+Gigabytes&amp;author=IH+Witten&amp;author=A+Moffat&amp;author=TC+Bell&amp;publication_year=1999&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R5">5. <span class="element-citation">Porter MF. An algorithm for suffix stripping. <span><span class="ref-journal">Program. </span>1980;<span class="ref-vol">14</span>:130–137.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Program&amp;title=An+algorithm+for+suffix+stripping&amp;author=MF+Porter&amp;volume=14&amp;publication_year=1980&amp;pages=130-137&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R6">6. <span class="element-citation">Harman D. How effective is suffixing? <span><span class="ref-journal">J Am Soc Inform Sci. </span>1991;<span class="ref-vol">42</span>:7–15.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=J+Am+Soc+Inform+Sci&amp;title=How+effective+is+suffixing?&amp;author=D+Harman&amp;volume=42&amp;publication_year=1991&amp;pages=7-15&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R7">7. <span class="element-citation">Xu J, Croft WB. Corpus-based stemming using co-occurrence of word variants. <span><span class="ref-journal">ACM Trans Inform Syst. </span>1979;<span class="ref-vol">16</span>:61–81.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=ACM+Trans+Inform+Syst&amp;title=Corpus-based+stemming+using+co-occurrence+of+word+variants&amp;author=J+Xu&amp;author=WB+Croft&amp;volume=16&amp;publication_year=1979&amp;pages=61-81&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R8">8. <span class="element-citation">Nadkarni PM, Chen RS, Brandt CA. UMLS concept indexing for production databases: a feasibility study. <span><span class="ref-journal">J Am Med Inform Assoc. </span>2001;<span class="ref-vol">8</span>:80–91.</span> <span class="nowrap">[<a class="int-reflink" href="/pmc/articles/PMC134593/">PMC free article</a>]</span> [<a href="/pubmed/11141514" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=J+Am+Med+Inform+Assoc&amp;title=UMLS+concept+indexing+for+production+databases:+a+feasibility+study&amp;author=PM+Nadkarni&amp;author=RS+Chen&amp;author=CA+Brandt&amp;volume=8&amp;publication_year=2001&amp;pages=80-91&amp;pmid=11141514&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R9">9. <span class="element-citation">Elkin PL, Cimino JJ, Lowe HJ, Aronow DB, Payne TH, Pincett PS, et al.  Mapping to MESH: the art of trapping MESH equivalence from within narrative text. Proc Symposium on Computer Applications in Medical Care; 1988. pp. 185–190. <span class="nowrap">[<a href="https://scholar.google.com/scholar?q=Elkin+PL+Cimino+JJ+Lowe+HJ+Aronow+DB+Payne+TH+Pincett+PS+Mapping+to+MESH:+the+art+of+trapping+MESH+equivalence+from+within+narrative+text+Proc+Symposium+on+Computer+Applications+in+Medical+Care+1988+185+190+" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R10">10. <span class="element-citation">Aronson A, Rindflesch T, Browne A. Exploiting a large thesaurus for information retrieval. Proceedings of the RIAO; 1994. pp. 197–216. <span class="nowrap">[<a href="https://scholar.google.com/scholar?q=Aronson+A+Rindflesch+T+Browne+A+Exploiting+a+large+thesaurus+for+information+retrieval+Proceedings+of+the+RIAO+1994+197+216+" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R11">11. <span class="element-citation">Aronson AR, Rindflesch TC. Query expansion using the UMLS Metathesaurus. Proceedings/AMIA Annual Fall Symposium; 1997. pp. 485–489. <span class="nowrap">[<a class="int-reflink" href="/pmc/articles/PMC2233565/">PMC free article</a>]</span> [<a href="/pubmed/9357673" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar?q=Aronson+AR+Rindflesch+TC+Query+expansion+using+the+UMLS+Metathesaurus+Proceedings/AMIA+Annual+Fall+Symposium+1997+485+489+" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R12">12. <span class="element-citation">Rindflesch TC, Aronson AR. Ambiguity resolution while mapping free text to the UMLS Metathesaurus. Proceedings the Annual Symposium on Computer Applications in Medical Care; 1994. pp. 240–244. <span class="nowrap">[<a class="int-reflink" href="/pmc/articles/PMC2247780/">PMC free article</a>]</span> [<a href="/pubmed/7949927" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar?q=Rindflesch+TC+Aronson+AR+Ambiguity+resolution+while+mapping+free+text+to+the+UMLS+Metathesaurus+Proceedings+the+Annual+Symposium+on+Computer+Applications+in+Medical+Care+1994+240+244+" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R13">13. <span class="element-citation">Masys D. Linking microarray data to the literature (Editorial) <span><span class="ref-journal">Nature Genet. </span>2001;<span class="ref-vol">27</span>:9–10.</span> [<a href="/pubmed/11326264" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Nature+Genet&amp;title=Linking+microarray+data+to+the+literature+(Editorial)&amp;author=D+Masys&amp;volume=27&amp;publication_year=2001&amp;pages=9-10&amp;pmid=11137987&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R14">14. <span class="element-citation">Mutalik P, Deshpande A, Nadkarni P. Use of general-purpose negation detection to augment concept indexing of medical documents: a quantitative study using the UMLS. <span><span class="ref-journal">J Am Med Inform Assoc. </span>2001;<span class="ref-vol">8</span>:598–609.</span> <span class="nowrap">[<a class="int-reflink" href="/pmc/articles/PMC130070/">PMC free article</a>]</span> [<a href="/pubmed/11687566" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=J+Am+Med+Inform+Assoc&amp;title=Use+of+general-purpose+negation+detection+to+augment+concept+indexing+of+medical+documents:+a+quantitative+study+using+the+UMLS&amp;author=P+Mutalik&amp;author=A+Deshpande&amp;author=P+Nadkarni&amp;volume=8&amp;publication_year=2001&amp;pages=598-609&amp;pmid=11687566&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R15">15. <span class="element-citation">Williams JH, Perriens MP. Automated full text indexing and searching systems. IBM Information Systems Symposium; Washington, DC. 1968. pp. 335–350. <span class="nowrap">[<a href="https://scholar.google.com/scholar?q=Williams+JH+Perriens+MP+Automated+full+text+indexing+and+searching+systems+IBM+Information+Systems+Symposium+Washington,+DC+1968+335+350+" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R16">16. <span class="element-citation">Sparck Jones K. A statistical interpretation of term specificity and its application in retrieval. <span><span class="ref-journal">J Documentation. </span>1972;<span class="ref-vol">28</span>:11–21.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=J+Documentation&amp;title=A+statistical+interpretation+of+term+specificity+and+its+application+in+retrieval&amp;author=K+Sparck+Jones&amp;volume=28&amp;publication_year=1972&amp;pages=11-21&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R17">17. <span class="element-citation">Sparck-Jones K, Walter S, Robertson SE. Information retrieval: development and comparative experiments (Part I) <span><span class="ref-journal">Inform Proc Manage. </span>2000;<span class="ref-vol">36</span>:779–808.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Inform+Proc+Manage&amp;title=Information+retrieval:+development+and+comparative+experiments+(Part+I)&amp;author=K+Sparck-Jones&amp;author=S+Walter&amp;author=SE+Robertson&amp;volume=36&amp;publication_year=2000&amp;pages=779-808&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R18">18. <span class="element-citation">Sparck-Jones K, Walter S, Robertson SE. Information retrieval: development and comparative experiments (Part 2) <span><span class="ref-journal">Inform Proc Manage. </span>2000;<span class="ref-vol">36</span>:809–840.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Inform+Proc+Manage&amp;title=Information+retrieval:+development+and+comparative+experiments+(Part+2)&amp;author=K+Sparck-Jones&amp;author=S+Walter&amp;author=SE+Robertson&amp;volume=36&amp;publication_year=2000&amp;pages=809-840&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R19">19. <span class="element-citation">Google Inc. <span class="ref-journal">Google: Technology Overview.</span> 2001.  <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?title=Google:+Technology+Overview&amp;publication_year=2001&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R20">20. <span class="element-citation">Marshall E. Medline searches turn up cases of suspected plagiarism (News) <span><span class="ref-journal">Science. </span>1998;<span class="ref-vol">279</span>:473–474.</span> [<a href="/pubmed/9454343" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Science&amp;title=Medline+searches+turn+up+cases+of+suspected+plagiarism+(News)&amp;author=E+Marshall&amp;volume=279&amp;publication_year=1998&amp;pages=473-474&amp;pmid=9454343&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R21">21. <span class="element-citation">OMIM. <span class="ref-journal">Online Mendelian Inheritance in Man.</span> McKusick Nathans Institute for Genetic Medicine, Johns Hopkins University; Baltimore, MD: National Center for Biotechnology Information, National Library of Medicine; Bethesda, MD: 2001.  <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?title=Online+Mendelian+Inheritance+in+Man&amp;publication_year=2001&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R22">22. <span class="element-citation">Ley K, Brewer K, Moton A. A web-based research tool for functional genomics of the microcirculation: the leukocyte adhesion cascade. <span><span class="ref-journal">Microcirculation. </span>1999;<span class="ref-vol">6</span>:259–265.</span> [<a href="/pubmed/10654277" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Microcirculation&amp;title=A+web-based+research+tool+for+functional+genomics+of+the+microcirculation:+the+leukocyte+adhesion+cascade&amp;author=K+Ley&amp;author=K+Brewer&amp;author=A+Moton&amp;volume=6&amp;publication_year=1999&amp;pages=259-265&amp;pmid=10654277&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R23">23. <span class="element-citation">Achard F, Vayssix G, Dessen P, Barillot E. Virgil database for rich links (1999 update) <span><span class="ref-journal">Nucl Acids Res. </span>1999;<span class="ref-vol">27</span>:113–114.</span> <span class="nowrap">[<a class="int-reflink" href="/pmc/articles/PMC148108/">PMC free article</a>]</span> [<a href="/pubmed/9847153" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Nucl+Acids+Res&amp;title=Virgil+database+for+rich+links+(1999+update)&amp;author=F+Achard&amp;author=G+Vayssix&amp;author=P+Dessen&amp;author=E+Barillot&amp;volume=27&amp;publication_year=1999&amp;pages=113-114&amp;pmid=9847153&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R24">24. <span class="element-citation">Rebhan M, Chalifa-Casp iV, Prilusky J, Lancet D. GeneCards: a novel functional genomics compendium with automated data mining and query reformulation support. <span><span class="ref-journal">Bioinformatics. </span>1998;<span class="ref-vol">14</span>:656–664.</span> [<a href="/pubmed/9789091" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Bioinformatics&amp;title=GeneCards:+a+novel+functional+genomics+compendium+with+automated+data+mining+and+query+reformulation+support&amp;author=M+Rebhan&amp;author=iV+Chalifa-Casp&amp;author=J+Prilusky&amp;author=D+Lancet&amp;volume=14&amp;publication_year=1998&amp;pages=656-664&amp;pmid=9789091&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R25">25. <span class="element-citation">Wu S, Manber U. Fast text searching allowing errors. <span><span class="ref-journal">Commun ACM. </span>1992;<span class="ref-vol">35</span>:83–91.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Commun+ACM&amp;title=Fast+text+searching+allowing+errors&amp;author=S+Wu&amp;author=U+Manber&amp;volume=35&amp;publication_year=1992&amp;pages=83-91&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R26">26. <span class="element-citation">Masys D, Welsh J, Lynn Fink J, MG, Klacansky I, Corbeil J. Use of keyword hierarchies to interpret gene expression patterns. <span><span class="ref-journal">Bioinformatics. </span>2001;<span class="ref-vol">17</span>:319–326.</span> [<a href="/pubmed/11301300" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Bioinformatics&amp;title=Use+of+keyword+hierarchies+to+interpret+gene+expression+patterns&amp;author=D+Masys&amp;author=J+Welsh&amp;author=J+Lynn+Fink&amp;author=+MG&amp;author=I+Klacansky&amp;volume=17&amp;publication_year=2001&amp;pages=319-326&amp;pmid=11301300&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R27">27. <span class="element-citation">National Center for Biotechnology Information. <span class="ref-journal">PubMed help.</span> 2001.  <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?title=PubMed+help&amp;publication_year=2001&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R28">28. <span class="element-citation">Tanabe L, Scherf U, Smith L, Lee J, Hunter L, Weinstein J. MedMiner: an Internet text-mining tool for biomedical information, with application to gene expression profiling. <span><span class="ref-journal">Biotechniques. </span>1999;<span class="ref-vol">27</span>:1210–1217.</span> [<a href="/pubmed/10631500" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Biotechniques&amp;title=MedMiner:+an+Internet+text-mining+tool+for+biomedical+information,+with+application+to+gene+expression+profiling&amp;author=L+Tanabe&amp;author=U+Scherf&amp;author=L+Smith&amp;author=J+Lee&amp;author=L+Hunter&amp;volume=27&amp;publication_year=1999&amp;pages=1210-1217&amp;pmid=10631500&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R29">29. <span class="element-citation">Rindflesch T, Hunter L, Aronson A. Mining molecular binding terminology from biomedical text. AMIA Fall Symposium; 1999. pp. 127–31. <span class="nowrap">[<a class="int-reflink" href="/pmc/articles/PMC2232739/">PMC free article</a>]</span> [<a href="/pubmed/10566334" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar?q=Rindflesch+T+Hunter+L+Aronson+A+Mining+molecular+binding+terminology+from+biomedical+text+AMIA+Fall+Symposium+1999+127+31+" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R30">30. <span class="element-citation">Rindflesch T, Tanabe L, Weinstein J, Hunter L. EDGAR: extraction of drugs, genes and relations from the biomedical literature. Pacific Symposium on Biocomputing; Honolulu, Hawaii. 2000. pp. 517–528. <span class="nowrap">[<a class="int-reflink" href="/pmc/articles/PMC2709525/">PMC free article</a>]</span> [<a href="/pubmed/10902199" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar?q=Rindflesch+T+Tanabe+L+Weinstein+J+Hunter+L+EDGAR:+extraction+of+drugs,+genes+and+relations+from+the+biomedical+literature+Pacific+Symposium+on+Biocomputing+Honolulu,+Hawaii+2000+517+528+" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R31">31. <span class="element-citation">Swanson D, Smalheiser N. An interactive system for finding complementary literatures: a stimulus to scientific discovery. <span><span class="ref-journal">Artif Intell. </span>1997;<span class="ref-vol">91</span>:183–203.</span> <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Artif+Intell&amp;title=An+interactive+system+for+finding+complementary+literatures:+a+stimulus+to+scientific+discovery&amp;author=D+Swanson&amp;author=N+Smalheiser&amp;volume=91&amp;publication_year=1997&amp;pages=183-203&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R32">32. <span class="element-citation">Finn R. Program uncovers hidden connections in the literature. <span><span class="ref-journal">The Scientist. </span>1998;<span class="ref-vol">12</span></span>  <a href="http://www.the-scientist.com" data-ga-action="click_feat_suppl" ref="reftype=extlink&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=External%7CLink%7CURI" target="_blank">www.the-scientist.com</a>.</span></div><div class="ref-cit-blk half_rhythm" id="R33">33. <span class="element-citation">Swanson D. Migraine and magnesium: eleven neglected connections. <span><span class="ref-journal">Perspect Biol Med. </span>1988;<span class="ref-vol">31</span>:526–557.</span> [<a href="/pubmed/3075738" target="pmc_ext" ref="reftype=pubmed&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord">PubMed</a>] <span class="nowrap">[<a href="https://scholar.google.com/scholar_lookup?journal=Perspect+Biol+Med&amp;title=Migraine+and+magnesium:+eleven+neglected+connections&amp;author=D+Swanson&amp;volume=31&amp;publication_year=1988&amp;pages=526-557&amp;pmid=3075738&amp;" target="pmc_ext" ref="reftype=other&amp;article-id=3137130&amp;issue-id=198522&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CLink%7CGoogle%20Scholar">Google Scholar</a>]</span></span></div></div></div></div><!--post-content--></div>
            
            
        
            
        </div>
        <!-- Book content -->
    </div>
    
    <div id="rightcolumn" class="four_col col last">
        <!-- Custom content above discovery portlets -->
        <div class="col6">
            
        </div>
        
        <div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><div class="format-menu"><h2>Formats:</h2><ul><li class="selected">Article</li> | <li><a href="/pmc/articles/PMC3137130/?report=reader">PubReader</a></li> | <li class="epub-link"><a href="/pmc/articles/PMC3137130/epub/">ePub (beta)</a></li> | <li><a href="/pmc/articles/PMC3137130/pdf/nihms303260.pdf">PDF (65K)</a></li> | <li><a href="#" data-citationid="PMC3137130" class="citationexporter ctxp">Citation</a></li></ul></div></div><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="share-buttons"><h2>Share</h2><ul><li class="facebook"><a href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3137130%2F"><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047626" alt="Share on Facebook" />
                             Facebook
                        </a></li><li class="twitter"><a href="https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3137130%2F&amp;text=An%20introduction%20to%20information%20retrieval%3A%20applications%20in%20genomics"><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047627" alt="Share on Twitter" />
                             Twitter
                        </a></li><li class="gplus"><a href="https://plus.google.com/share?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3137130%2F"><img src="//static.pubmed.gov/portal/portal3rc.fcgi/4160049/img/4047628" alt="Share on Google Plus" />
                             Google+
                        </a></li></ul></div>
        
        <div id="ajax-portlets" data-pmid="12049181" data-aiid="3137130" data-aid="3137130" data-iid="198522" data-domainid="319" data-domain="nihpa" data-accid="PMC3137130" data-md5="7c4deeedf52ff71cbe689f38303cc0c3"></div>
                
        <!-- Custom content below discovery portlets -->
        <div class="col7">
            
        </div>
    </div>
</div>

<!-- Custom content after all -->
<div class="col8">
    
</div>
<div class="col9">
    
</div>

<script src="/corehtml/pmc/js/jquery.scrollTo-1.4.2.js"></script>
<script>
    (function($){
        $('.skiplink').each(function(i, item){
            var href = $($(item).attr('href'));
            href.attr('tabindex', '-1').addClass('skiptarget'); // ensure the target can receive focus
            $(item).on('click', function(event){
                event.preventDefault();
                $.scrollTo(href, 0, {
                    onAfter: function(){
                        href.focus();
                    }
                });
            });
        });
    })(jQuery);
</script>



<div id="body-link-poppers"></div>
                        </div>
                        <div class="bottom">
                            
                            <div id="NCBIFooter_dynamic">
    <a id="help-desk-link" class="help_desk" href="" target="_blank">Support Center</a>
    <a id="help-desk-link" class="help_desk" href="https://support.ncbi.nlm.nih.gov/ics/support/KBList.asp?Time=2020-02-24T00:11:07-05:00&amp;Snapshot=%2Fprojects%2FPMC%2FPMCViewer@4.46&amp;Host=ptpmc101&amp;ncbi_phid=8A1A8E52E53566610000000005600560&amp;ncbi_session=CE8C90BAE2FBC491_2129SID&amp;from=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3137130%2F&amp;Db=pmc&amp;folderID=132&amp;Ncbi_App=pmc&amp;Page=literature&amp;style=classic&amp;deptID=28049" target="_blank">Support Center</a>
    
</div>

                            <div class="footer" id="footer">
    
    <div class="subfooter"> </div><script type="text/javascript" src="/portal/portal3rc.fcgi/static/js/preloaderWidget.js"> </script>
    <div id="external-disclaimer" class="offscreen_noflow">
        External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.
    </div>    
    <div id="ncbifooter" class="contact_info">      
        <div id="footer-contents-right">
            <div id="nlm_thumb_logo">
                <a href="https://www.nlm.nih.gov" title="NLM">NLM</a>
            </div>
            <div id="nih_thumb_logo">
                <a href="https://www.nih.gov" title="NIH">NIH</a>
            </div>
            <div id="hhs_thumb_logo">
                <a href="https://www.hhs.gov" title="DHHS">DHHS</a>
            </div>
            <div id="usagov_thumb_logo">
                <a href="https://www.usa.gov" title="USA.gov">USA.gov</a>
            </div>         
        </div>
        
        <div id="footer-contents-left">
            <p class="address vcard">
                <span class="url">
                    <a class="fn url newdomain" href="https://www.ncbi.nlm.nih.gov">National Center for
                        Biotechnology Information</a>,
                </span> <span class="org url newdomain"><a href="https://www.nlm.nih.gov/">U.S. National Library of Medicine</a></span>
                <span class="adr">
                    <span class="street-address">8600 Rockville Pike</span>, <span class="locality">Bethesda</span>
                    <span class="region">MD</span>, <span class="postal-code">20894</span>
                    <span class="country-name">USA</span>
                </span>
            </p>
            
            <a href="/home/about/policies.shtml">Policies and Guidelines</a> | <a href="/home/about/contact.shtml">Contact</a>
        </div>
    </div>
    <script type="text/javascript" src="/portal/portal3rc.fcgi/rlib/js/InstrumentOmnitureBaseJS/InstrumentNCBIConfigJS/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js?v=1"> </script>    
    <script type="text/javascript" src="/portal/portal3rc.fcgi/static/js/hfjs2.js"> </script>
</div>
                        </div>
                    </div>
                    <!--/.page-->
                </div>
                <!--/.wrap-->
            </div><!-- /.twelve_col -->
        </div>
        <!-- /.grid -->

        <span class="PAFAppResources"></span>
        
        <!-- BESelector tab -->
        
        
        
        <noscript><img alt="statistics" src="/stat?jsdisabled=true&amp;ncbi_db=pmc&amp;ncbi_pdid=article&amp;ncbi_acc=&amp;ncbi_domain=nihpa&amp;ncbi_report=record&amp;ncbi_type=fulltext&amp;ncbi_objectid=&amp;ncbi_pcid=/articles/PMC3137130/&amp;ncbi_app=pmc" /></noscript>
        
        
        <!-- usually for JS scripts at page bottom -->
        <!--<component id="PageFixtures" label="styles"></component>-->
    

<!-- CE8C90BAE2FBC491_2129SID /projects/PMC/PMCViewer@4.46 ptpmc101 v4.1.r585844 Mon, May 06 2019 02:53:16 -->

<script type="text/javascript" src="//static.pubmed.gov/portal/portal3rc.fcgi/4160049/js/3879255/4121861/3818874/4168176/3821238/4117325/4087685/4072593/4076480/3921943/4105668/4065628.js" snapshot="pmc"></script></body>
</html>